backends/qualcomm/quantizer/custom_annotation.py

```diff
@@ -118,3 +120,35 @@ def annotate_matmul_input1(node: Node, quantization_config: QuantizationConfig):
                 if "SDPA" in full_qualified_name:
                     annotate_matmul(node, quantization_config_16a8w)
                     annotate_matmul_input1(node.args[1], quantization_config_8a8w)
+
+
+def get_custom_quant_ios_dtype(
+    cache_shape: torch.Size,
+    node: torch.fx.Node,
+    kv_dtype=torch.uint8,
+    sharding_dtype=torch.uint16,
+):
+    """
+    This function is specific for llama inputs and outputs
```

Comment:
I'm trying to understand why it is specific to llama inputs/outputs. Is it because of the sharding of the model? Like the output of the first sharding doesn't need to dequant and the input of the second sharding doesn't need quant node?

Reply:
Let me try to make it clear.
This function is to clean the redundant Q/DQ nodes such as KV I/O and intermediate tensors between the sharding as you mentioned.
In original flow, we will quantize KV input and dequantize KV output every inference. 
In fact, we don’t need to do this, we can directly output the quantized KV and put it into the model for the next inference.